presto-hive/src/main/java/com/facebook/presto/hive/DataSinkStats.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import io.airlift.stats.DistributionStat;
+import org.weakref.jmx.Managed;
+import org.weakref.jmx.Nested;
+
+public class DataSinkStats
+{
+    private final DistributionStat flushedPageSizesInBytes = new DistributionStat();
+
+    @Managed
+    @Nested
+    public DistributionStat getFlushedPageSizesInBytes()
+    {
+        return flushedPageSizesInBytes;
+    }
+
+    public void addFlushedPageSizesInBytes(long bytes)
+    {
+        flushedPageSizesInBytes.add(bytes);
```

Comment:
maybe rename this to "inputPageSizeInBytes".  I was confused because I thought this was measuring the written data, not the distribution of pages sent to the writer.